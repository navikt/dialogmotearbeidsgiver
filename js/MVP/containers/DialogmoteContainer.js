import React from 'react';
import { Normaltekst, Sidetittel } from 'nav-frontend-typografi';
import Lenke from 'nav-frontend-lenker';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Brodsmuler from '../../components/Brodsmuler';
import { dialogmoteBreadcrumb } from '../globals/paths';

const WrappperStyled = styled.div`
  display: flex;
  justify-content: center;
  background-color: #f1f1f1;
  padding: 32px;
`;

const ContentStyled = styled.div`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  align-items: stretch;
  max-width: 640px;
`;

const HeaderStyled = styled.header`
  margin: 32px 0;
  text-align: center;
`;

const BottomInfoStyled = styled.section`
  text-align: center;
  margin-top: auto;
`;

const texts = {
  bottomText: 'Vi bruker opplysningene også til å gjøre selve tjenesten bedre.',
  bottomUrl: 'Les mer om hvordan NAV behandler personopplysninger.',
};

const DialogmoteContainer = ({ title, breadcrumb = dialogmoteBreadcrumb, children }) => {
  return (
    <WrappperStyled>
      <ContentStyled>
        <Brodsmuler brodsmuler={breadcrumb} />
        <HeaderStyled>
          <Sidetittel>{title}</Sidetittel>
        </HeaderStyled>
        {children}
        <BottomInfoStyled>
          <Normaltekst>{texts.bottomText}</Normaltekst>
          <Lenke>{texts.bottomUrl}</Lenke>
        </BottomInfoStyled>
      </ContentStyled>
    </WrappperStyled>
  );
};

DialogmoteContainer.propTypes = { title: PropTypes.string, children: PropTypes.node, breadcrumb: PropTypes.array };

export default DialogmoteContainer;
