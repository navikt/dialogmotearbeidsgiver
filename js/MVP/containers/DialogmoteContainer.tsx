import React, { ReactElement, ReactNode } from 'react';
import { Normaltekst, Sidetittel } from 'nav-frontend-typografi';
import { useHistory } from 'react-router-dom';
import styled from 'styled-components';
import Brodsmuler, { BrodsmuleProps } from '../../components/Brodsmuler';
import { statiskeURLer } from '../globals/paths';
import Lenke from 'nav-frontend-lenker';
import { eventNames } from '@/amplitude/events';
import { Tilbakeknapp } from 'nav-frontend-ikonknapper';
import { trackOnClick } from '@/amplitude/amplitude';

const WrapperStyled = styled.div`
  display: flex;
  justify-content: center;
  background-color: #f1f1f1;
  padding: 32px;
`;

const ContentStyled = styled.div`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  align-items: stretch;
  max-width: 640px;
`;

const HeaderStyled = styled.header`
  margin: 32px 0;
  text-align: center;
`;

const TilbakeknappStyled = styled(Tilbakeknapp)`
  width: 108px;
  margin-bottom: 32px;
`;

const BottomInfoStyled = styled.section`
  text-align: center;
  margin-top: auto;
`;

const texts = {
  bottomText: 'Vi bruker opplysningene også til å gjøre selve tjenesten bedre.',
  bottomUrl: 'Les mer om hvordan NAV behandler personopplysninger.',
};

interface Props {
  title?: string;
  children: ReactNode;
  displayTilbakeknapp?: boolean;
  breadcrumb: BrodsmuleProps[];
}

const DialogmoteContainer = ({ title, breadcrumb, displayTilbakeknapp = false, children }: Props): ReactElement => {
  const history = useHistory();

  return (
    <WrapperStyled>
      <ContentStyled>
        <Brodsmuler brodsmuler={breadcrumb} />
        {title && (
          <HeaderStyled>
            <Sidetittel>{title}</Sidetittel>
          </HeaderStyled>
        )}
        {children}
        {displayTilbakeknapp && <TilbakeknappStyled onClick={history.goBack} />}
        <BottomInfoStyled>
          <Normaltekst>{texts.bottomText}</Normaltekst>
          <Lenke
            href={statiskeURLer.PERSONVERN_URL}
            onClick={() => trackOnClick(eventNames.behandlePersonopplysninger)}
          >
            {texts.bottomUrl}
          </Lenke>
        </BottomInfoStyled>
      </ContentStyled>
    </WrapperStyled>
  );
};

export default DialogmoteContainer;
