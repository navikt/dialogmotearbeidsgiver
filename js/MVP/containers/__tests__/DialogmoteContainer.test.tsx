import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import DialogmoteContainer from '../DialogmoteContainer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

describe('DialogmoteContainer', () => {
  test('should render', () => {
    const { container } = render(
      <DialogmoteContainer title="Test title" breadcrumb={[]}>
        <div>Test div</div>
      </DialogmoteContainer>
    );

    expect(container).toHaveTextContent('Test div');
  });

  test('should render tilbakeknapp and invoke goBack on click', () => {
    const goBackMock = jest.fn();
    const historyStub = createMemoryHistory();
    const history = { ...historyStub, goBack: goBackMock };

    render(
      <Router history={history}>
        <DialogmoteContainer title="Test title" breadcrumb={[]} displayTilbakeknapp={true}>
          <div>Test div</div>
        </DialogmoteContainer>
      </Router>
    );

    const tilbakeknapp = screen.getByRole('button', { name: 'Tilbake' });
    expect(tilbakeknapp).toBeInTheDocument();

    fireEvent.click(tilbakeknapp);
    expect(goBackMock).toHaveBeenCalled();
  });
});
