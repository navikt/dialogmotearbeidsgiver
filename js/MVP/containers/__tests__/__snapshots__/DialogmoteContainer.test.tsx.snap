// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogmoteContainer should render 1`] = `
<div>
  <div
    class="sc-bdvvtL dqnhhB"
  >
    <div
      class="sc-gsDKAQ cCAVmz"
    >
      <nav
        aria-label="Du er her: "
        class="brodsmuler"
      >
        <img
          alt="Du"
          class="brodsmuler__ikon"
          src="[object Object]"
        />
        <div
          class="brodsmuler__smuler"
        />
      </nav>
      <header
        class="sc-dkPtRN dCgkxL"
      >
        <h1
          class="typo-sidetittel"
        >
          Test title
        </h1>
      </header>
      <div>
        Test div
      </div>
      <section
        class="sc-eCImPb epfClQ"
      >
        <p
          class="typo-normal"
        >
          Vi bruker opplysningene også til å gjøre selve tjenesten bedre.
        </p>
        <a
          class="lenke"
          href="http://www.nav.no/personvern"
        >
          Les mer om hvordan NAV behandler personopplysninger.
        </a>
      </section>
    </div>
  </div>
</div>
`;
